File: config/project.js
// Centralized project configuration
// This file reads from .project and provides configuration to all parts of the application

const fs = require('fs');
const path = require('path');

// Read .project file
function loadProjectConfig() {
  const projectFile = path.join(__dirname, '..', '.project');
  const content = fs.readFileSync(projectFile, 'utf8');
  const config = {};
  
  content.split('\n').forEach(line => {
    const [key, value] = line.split('=');
    if (key && value) {
      config[key.trim()] = value.trim();
    }
  });
  
  return config;
}

const projectConfig = loadProjectConfig();

// Export configuration
module.exports = {
  // Project Info
  projectName: projectConfig.PROJECT_NAME || 'RapidTriageME',
  repositoryUrl: projectConfig.REPOSITORY_URL || 'https://github.com/YarlisAISolutions/rapidtriageME',
  
  // Repository parts
  githubOrg: projectConfig.REPOSITORY_URL?.split('/')[3] || 'YarlisAISolutions',
  githubRepo: projectConfig.REPOSITORY_URL?.split('/')[4] || 'rapidtriageME',
  
  // Infrastructure
  provider: projectConfig.PROVIDER || 'CLOUDFLARE',
  region: projectConfig.REGION || 'GLOBAL',
  domain: projectConfig.DOMAIN || 'rapidtriage.me',
  
  // Documentation
  docsUrl: `https://docs.${projectConfig.DOMAIN || 'rapidtriage.me'}`,
  
  // Description
  description: projectConfig.DESCRIPTION || 'Remote Browser Tools MCP Platform',
  
  // Utility functions
  getGitCloneUrl: function() {
    return `${this.repositoryUrl}.git`;
  },
  
  getIssuesUrl: function() {
    return `${this.repositoryUrl}/issues`;
  },
  
  getRawContentUrl: function(branch = 'main', filepath = '') {
    return `https://raw.githubusercontent.com/${this.githubOrg}/${this.githubRepo}/${branch}/${filepath}`;
  }
};

File: config/project.py
#!/usr/bin/env python3
"""
Centralized project configuration for Python/MkDocs
Reads from .project file and provides configuration
"""

import os
from pathlib import Path

def load_project_config():
    """Load configuration from .project file"""
    project_file = Path(__file__).parent.parent / '.project'
    config = {}
    
    if project_file.exists():
        with open(project_file, 'r') as f:
            for line in f:
                if '=' in line:
                    key, value = line.strip().split('=', 1)
                    config[key] = value
    
    return config

# Load configuration
_config = load_project_config()

# Project configuration
PROJECT_NAME = _config.get('PROJECT_NAME', 'RapidTriageME')
REPOSITORY_URL = _config.get('REPOSITORY_URL', 'https://github.com/YarlisAISolutions/rapidtriageME')
PROVIDER = _config.get('PROVIDER', 'CLOUDFLARE')
REGION = _config.get('REGION', 'GLOBAL')
DOMAIN = _config.get('DOMAIN', 'rapidtriage.me')
DESCRIPTION = _config.get('DESCRIPTION', 'Remote Browser Tools MCP Platform')

# Derived values
GITHUB_ORG = REPOSITORY_URL.split('/')[3] if '/' in REPOSITORY_URL else 'YarlisAISolutions'
GITHUB_REPO = REPOSITORY_URL.split('/')[4] if '/' in REPOSITORY_URL else 'rapidtriageME'
DOCS_URL = f'https://docs.{DOMAIN}'
ISSUES_URL = f'{REPOSITORY_URL}/issues'

# MkDocs specific configuration
MKDOCS_CONFIG = {
    'site_name': f'{PROJECT_NAME} Documentation',
    'site_url': DOCS_URL,
    'repo_url': REPOSITORY_URL,
    'repo_name': f'{GITHUB_ORG}/{GITHUB_REPO}',
    'site_description': f'{DESCRIPTION} - Complete Documentation',
    'site_author': GITHUB_ORG,
    'edit_uri': 'edit/main/docs/',
    'copyright': f'&copy; 2025 <a href="https://yarlis.ai" target="_blank" rel="noopener">{GITHUB_ORG}</a>'
}

if __name__ == '__main__':
    # Print configuration when run directly
    print("Project Configuration:")
    print(f"  Name: {PROJECT_NAME}")
    print(f"  Repository: {REPOSITORY_URL}")
    print(f"  Domain: {DOMAIN}")
    print(f"  Docs URL: {DOCS_URL}")
    print(f"  GitHub Org: {GITHUB_ORG}")
    print(f"  GitHub Repo: {GITHUB_REPO}")